lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -418,17 +418,39 @@ public function commit($entity = null)
 
         $this->dispatchPostFlushEvent();
 
+        $this->postCommitClear($entity);
+    }
+
+    /**
+     * @param null|object|array $entity
+     */
+    private function postCommitClear($entity = null) {
+
         // Clear up
         $this->entityInsertions =
         $this->entityUpdates =
         $this->entityDeletions =
         $this->extraUpdates =
-        $this->entityChangeSets =
         $this->collectionUpdates =
         $this->collectionDeletions =
         $this->visitedCollections =
-        $this->scheduledForSynchronization =
         $this->orphanRemovals = array();
+
+        if ($entity === null) {
+            $this->entityChangeSets = $this->scheduledForSynchronization = array();
+        } elseif (is_object($entity)) {
```

Comment:
Can only be an array or an object, therefore doesn't need checking for `is_object` if the `is_array` check comes before this one